rule/src/main/java/com/buschmais/jqassistant/core/rule/impl/reader/IndentHelper.java

```diff
@@ -0,0 +1,124 @@
+package com.buschmais.jqassistant.core.rule.impl.reader;
+
+public class IndentHelper {
+    /**
+     * This class make the String more effective and understandable. by removing the extra spaces,lines and tabs.
+     * even tho , there is some exceptions where if the tabs are not in order the whole class won't work!
+     * NOTE : Check the TextUtilTest.java to see the examples
+     *
+     * @param text
+     * @return edited text without the unused whitespaces
+     */
+    public static String removeIndent(String text) {
+        if (text == null) {
+            return null;
+        }
+        String textWithoutEmptyLines = removeEmptyLines(text);
+        String[] lines = textWithoutEmptyLines.split("\\n");
+        if (text.isBlank()) {
+            return textWithoutEmptyLines;
+        }
+        int indent = getIndent(lines);
+        return removeIndent(lines, indent);
+    }
+
+    /**
+     * Removes the specified indentation from each line in the provided array of lines. If it's needed.
+     * Removes the trailing and leading spaces from each lines
+     *
```

Comment:
Add a hint that only indent is considered which contains exactly the same characters across all lines ignoring blank lines.